apollo-portal/src/main/java/com/ctrip/framework/apollo/openapi/filter/ConsumerAuthenticationFilter.java

```diff
@@ -55,14 +74,33 @@ public void doFilter(ServletRequest req, ServletResponse resp, FilterChain chain
     HttpServletResponse response = (HttpServletResponse) resp;
 
     String token = request.getHeader(HttpHeaders.AUTHORIZATION);
+    ConsumerToken consumerToken = consumerAuthUtil.getConsumerToken(token);
 
-    Long consumerId = consumerAuthUtil.getConsumerId(token);
-
-    if (consumerId == null) {
+    if (null == consumerToken || consumerToken.getConsumerId() <= 0) {
       response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Unauthorized");
       return;
     }
 
+    Integer limitCount = consumerToken.getLimitCount();
+    if (limitCount == null) {
+      limitCount = 0;
+    }
+    if (portalConfig.isOpenApiLimitEnabled() && limitCount > 0) {
+      try {
+        ImmutablePair<Long, RateLimiter> rateLimiterPair = getOrCreateRateLimiterPair(token, limitCount);
+        long warmupToMillis = rateLimiterPair.getLeft() + WARMUP_MILLIS;
```

Comment:
In actual testing, guava RateLimiter has a problem of incorrect limit in the first second. Using `com.ctrip.framework.apollo.openapi.filter.ConsumerAuthenticationFilterTest#testRateLimitSuccessfully`, it can be found that there is always one request that is limited.